### ISSUE TITLE
torch.cuda.OutOfMemoryError: CUDA out of memory.  in SS2Dvm0

### ISSUE THREAD

**Issue by JianTao-zjut** [MzeroMiko/VMamba#338]
I have been using SS2Dv2 for forward inference, which works perfectly and consumes about 22GiB of GPU memory. However, after replacing it with SS2Dm0, I encountered the following error:

torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 2.72 GiB. GPU 0 has a total capacty of 47.54 GiB of which 2.14 GiB is free. Including non-PyTorch memory, this process has 45.40 GiB memory in use. Of the allocated memory 44.90 GiB is allocated by PyTorch, and 164.54 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

Additionally, I noticed assertion errors (related to getCvtOrder) when using Triton. As a workaround, I forced the inference to run with Torch instead of Triton.

My Questions:

Could this issue be related to switching from Triton to Torch for inference?
If the input data remains identical, what might cause SS2Dm0 to use more GPU memory compared to SS2Dv2?

Additional Information:

Framework: PyTorch
GPU Model: NVIDIA A6000
CUDA Version: 11.8

**Comment by MzeroMiko** [comment]
Yes, the torch for-loop version cost much more memory than triton or cuda version (as it need to save every variable in its computing graph to make autograd possible) and is also much slower.
